### ISSUE TITLE
Can't install nuget because of reference to System.Linq.dll

### ISSUE THREAD

**Issue by 0x53A** [dadhi/FastExpressionCompiler#70]
When i try to install the latest version:

```
Attempting to gather dependency information for package 'FastExpressionCompiler.1.7.0' with respect to project 'WebApiSanityChecks', targeting '.NETFramework,Version=v4.6'
Attempting to resolve dependencies for package 'FastExpressionCompiler.1.7.0' with DependencyBehavior 'Highest'
Resolving actions to install package 'FastExpressionCompiler.1.7.0'
Resolved actions to install package 'FastExpressionCompiler.1.7.0'
  GET https://api.nuget.org/v3-flatcontainer/fastexpressioncompiler/1.7.0/fastexpressioncompiler.1.7.0.nupkg
  OK https://api.nuget.org/v3-flatcontainer/fastexpressioncompiler/1.7.0/fastexpressioncompiler.1.7.0.nupkg 177ms
Installing FastExpressionCompiler 1.7.0.
Adding package 'FastExpressionCompiler.1.7.0' to folder 'C:\Progress\MyProjects\WebApiSanityChecks\packages'
Added package 'FastExpressionCompiler.1.7.0' to folder 'C:\Progress\MyProjects\WebApiSanityChecks\packages'
Install failed. Rolling back...
Package 'FastExpressionCompiler.1.7.0' does not exist in project 'WebApiSanityChecks'
Removing package 'FastExpressionCompiler.1.7.0' from folder 'C:\Progress\MyProjects\WebApiSanityChecks\packages'
Removed package 'FastExpressionCompiler.1.7.0' from folder 'C:\Progress\MyProjects\WebApiSanityChecks\packages'
Failed to add reference to 'System.Linq'. Please make sure that it is in the Global Assembly Cache.
========== Finished ==========
```

Looking in the .nuspec, it contains:

```xml
    <frameworkAssemblies>
      <frameworkAssembly assemblyName="Microsoft.CSharp" targetFramework=".NETFramework4.5" />
      <frameworkAssembly assemblyName="System" targetFramework=".NETFramework4.5" />
      <frameworkAssembly assemblyName="System.Linq" targetFramework=".NETFramework4.5" />
      <frameworkAssembly assemblyName="System.Linq.Expressions" targetFramework=".NETFramework4.5" />
      <frameworkAssembly assemblyName="System.Reflection" targetFramework=".NETFramework4.5" />
    </frameworkAssemblies>
```

At the least System.Linq is incorrect and should be removed, I am not sure about the others.

**Comment by dadhi** [comment]
Strange. Thanks for posting. I will look.

@0x53A, Could you check if it installed if you target 4.6 or higher?

**Comment by 0x53A** [comment]
I tried it with 4.5 and 4.6.

I think these assemblies may be installed by __.NET 3.5__ into the GAC, which I don't have.
Afaik with 4.5+, the classes are in __System.Core__. (Which is referenced by default, so you shouldn't need the whole block.)

Anyway, I opened the nupkg and removed the whole ``<frameworkAssemblies>``, now it works :)

**Comment by dadhi** [comment]
@0x53A, 

Hi again, thanks for bringing this issue. I did not pay enough attention to full .NET plat, just copied it from somewhere.

Strange no one else reported, but it seems the FEC mostly used as single .cs file added to project.

Better to learn from the [tops](https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/Newtonsoft.Json.csproj) and fix this.

**Comment by dadhi** [comment]
I have probably suggested this to @EamonNerbonne. Now, I would've also suggested to make all public FEC types internal to prevent any conflicts. Here is the [pwsh script](https://bitbucket.org/dadhi/dryioc/src/default/BuildScripts/MakeInternal.ps1) I am using in DryIoc for this matter.

Btw, thanks for info regarding VS versions, started to forget about 2015 :)

**Comment by dadhi** [comment]
Linking #58 here

**Comment by EamonNerbonne** [comment]
The downside of literal inclusion is that it's more work to maintain than a nuget dependency is.

It would, however, also solve the code-signing issue...

I'm not a fan of submodules, but maybe it's an option for this.

**Comment by 0x53A** [comment]
Thank you for fixing this.

I now think this is a nuget bug, because in a new console app (still VS 2015), I can install the old package (1.7.1), but in my original project I still can't with the error of the first post.

Both are C#, both target net460, I really hate nuget.

**Comment by dadhi** [comment]
I have released  [1.7.2](https://www.nuget.org/packages/FastExpressionCompiler/1.7.2) with the fix.

**Comment by 0x53A** [comment]
I still haven't been able to figure out the difference between both projects, but 1.7.2 works with both!

Thank you!
